src/components/blog/BlogSection.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react'
+import { StaticQuery, graphql } from 'gatsby'
+
+import Section from 'src/components/layout/Section'
+import Grid, { Col, Row } from 'src/components/layout/Grid'
+import { H2 } from 'src/components/text'
+import PostCard from 'src/components/blog/PostCard'
+
+const postsThatMatchAllTags = tags => ({
+  node: {
+    frontmatter: { tags: postTags },
+  },
+}) =>
+  tags &&
+  tags.every(constraint => postTags && postTags.some(tag => tag === constraint))
+
+const BlogSection = ({ tags = [] }) => (
+  <StaticQuery
+    query={graphql`
+      query getPosts($limit: Int = 1000) {
+        allMarkdownRemark(
+          filter: {
+            fields: { slug: { regex: "/(/blog/|/react/|/graphql/)/" } }
+          }
+          sort: { fields: [frontmatter___order], order: DESC }
+          limit: $limit
+        ) {
+          edges {
+            node {
+              fields {
+                slug
+              }
+              frontmatter {
+                title
+                imageUrl
+                tags
+              }
+              excerpt
+            }
+          }
+        }
+      }
+    `}
+    render={data => {
+      const posts = data.allMarkdownRemark.edges
+        .filter(postsThatMatchAllTags(tags))
+        .slice(0, 3)
+
+      if (!posts || !posts.length) {
+        return null
+      }
+
+      // TODO Paul style it
```

Comment:
Is this styling supposed to happen before merging?

Reply:
I would do the styling/ UX improvement after.  It now looks as it is on the blog page. It can be better but it's using components that are already in prod.